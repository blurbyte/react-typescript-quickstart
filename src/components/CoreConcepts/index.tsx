import * as React from 'react';

import SecondaryHeadline from '../SecondaryHeadline';
import Text from '../Text';
import TextList from '../TextList';

class CoreConcepts extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <section>
        <SecondaryHeadline>What will you learn?</SecondaryHeadline>
        <Text>By inspecting source code of this demo application you can get a good grasp of:</Text>
        <TextList>
          <TextList.Item>How to set up TypeScript development environment for React web app</TextList.Item>
          <TextList.Item>How to organize file structure of your project</TextList.Item>
          <TextList.Item>
            How to implement and test <strong>render props</strong> component &mdash; <em>src/components/Counter</em>{' '}
            and <em>src/components/ManageCounter</em>
          </TextList.Item>
          <TextList.Item>
            How to create <strong>compound</strong> component &mdash; <em>src/components/TextList</em>
          </TextList.Item>
          <TextList.Item>How to write unit and integration tests</TextList.Item>
          <TextList.Item>How to setup strightforward CI/CD pipeline with TravisCI and Surge</TextList.Item>
          <TextList.Item>
            How to create fully responsive layout with <strong>css grid</strong> and <strong>styled-components</strong>
          </TextList.Item>
        </TextList>
      </section>
    );
  }
}

export default CoreConcepts;
