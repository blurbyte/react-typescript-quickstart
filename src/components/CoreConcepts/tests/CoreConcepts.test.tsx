import * as React from 'react';
import { shallow } from 'enzyme';

import CoreConcepts from '../index';

describe('CoreConcepts', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CoreConcepts />);
    expect(wrapper).toMatchSnapshot();
  });
});
