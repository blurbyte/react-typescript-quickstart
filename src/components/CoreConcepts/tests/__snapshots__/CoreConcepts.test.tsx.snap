// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreConcepts renders correctly 1`] = `
<React.Fragment>
  <styled.h2>
    What will you learn?
  </styled.h2>
  <styled.p>
    By inspecting source code of this demo application you can get a good grasp of:
  </styled.p>
  <TextList>
    <Styled(styled.li)>
      How to set up TypeScript development environment for React web app
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to organize file structure of your project
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to implement and test 
      <strong>
        render props
      </strong>
       component — 
      <em>
        src/components/Counter
      </em>
       
      and 
      <em>
        src/components/ManageCounter
      </em>
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to create 
      <strong>
        compound
      </strong>
       component — 
      <em>
        src/components/TextList
      </em>
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to write unit and integration tests
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to setup strightforward CI/CD pipeline with TravisCI and Surge
    </Styled(styled.li)>
    <Styled(styled.li)>
      How to create fully responsive layout with 
      <strong>
        css grid
      </strong>
       and 
      <strong>
        styled-components
      </strong>
    </Styled(styled.li)>
  </TextList>
</React.Fragment>
`;
